Show documents and tasks on repository folders.

jone opened the pull request:
At the moment it is not possible to list all documents or all tasks for a specific repository folder. This is especially annoing when trying to figure out how many / which tasks are open in a specific area of the repository. As user I'm sometimes looking for a document but I'm not sure in which dossier it is stored, for example because I'm not responsible / used to this area of the repository.

## Solution:
Adding the tabs "documents" and "tasks" on the repository folders will resolve that problem and give the users more flexibility. It also helps users to navigate faster because there is no need to open the dossier first.

## Implementation details:
- Both tabs can be individually disabled with feature flags. The tabs are enabled by default and will be automatically enabled on update.
- The documents tab has a new dossier column; the task tab already had one.
- The actions were cleaned up so that only actions are provided which actually work from the repository folder context.

## Screenshots:
![bildschirmfoto 2018-01-16 um 14 45 27](https://user-images.githubusercontent.com/7469/34992859-cea96600-face-11e7-8139-048617b57a0f.png)
![bildschirmfoto 2018-01-16 um 14 44 38](https://user-images.githubusercontent.com/7469/34992862-d076fa2e-face-11e7-9bbf-a83368350a61.png)


## Discussion:
The feature has been discussed in the feedback forum: https://feedback.onegovgever.ch/t/dokumente-und-aufgabenuebersicht-auf-stufe-ordnungsposition-aktivieren/523/9

- **Performance:** There was the objection that this might by very imperformant on top-level repository folders on very large installations. Therefore the tabs can be disabled by feature flag.

- **Visual similarity with dossier:** When the tabs were initially designed we did not include those tabs on repository folders in order to have a visual unambiguousness between repository folders and dossiers. This is problem is not anymore valid since we have changed the design completely in recent years and we have added water marks in the background.

lukasgraf commented on opengever/repository/browser/tabs.py:
```diff
@@ -0,0 +1,28 @@
+class RepositoryFolderDocuments(Documents):
+
+    disabled_actions = (
+        'zip_selected',  # disabled because ZIP filename is invalid
```
What exactly does this comment mean here?

jone replied:
I tried to download a bunch of documents on the documents-tab of a repository folder. I got a ZIP file, but it's name was just `zip` and MacOS couldn't extract it. 🤔

jone replied:
I have updated the comment.

lukasgraf commented on opengever/repository/browser/tabs.py:
```diff
@@ -0,0 +1,28 @@
+
+    @property
+    def columns(self):
+        columns = list(super(RepositoryFolderDocuments, self).columns)
```
Hmm. Luckily `Document.columns` is immutable (tuple), not a list. Otherwise the `insert()` below would mutate the `columns` on the `Document` class. Maybe create a copy? It only takes one Schlaumeier to change `Documents.columns` into a list for whatever reason, and this will break in horrible ways 😉

jone replied:
I think I'm already making a copy by calling the `list()`-type 😜

lukasgraf replied:
You're right - for some reason, I was under the assumption, that `(list(already_a_list) is already_a_list) == True` (that Python would directly return the reference). Which is obviously false. 😜

lukasgraf commented on opengever/repository/tests/test_repositoryfolder_tabs.py:
```diff
@@ -9,12 +9,66 @@ class TestRepositoryFolderTabs(IntegrationTestCase):
+    def test_visible_actions(self, browser):
+        # Not all actions of the dossier-tab work on the repositoryfolder-tab.
+        # We are testing as Manager so that we can make sure that actions are
+        # filtered without programatically, without respecting permissions.
```
-`without programatically,` (I think?)

jone replied:
I have rewritten this part.

jone commented:
- Updated according to comments.
- I've also removed the tasks-tab, since it was actually wrong and not really used; it works well with the default tab.
- I have added a test for the tasks tab in order to reveal future problems with changes on the regular task tab, which have a negative effect to the repository folder tab.
